File: backend/mscliente/src/main/java/mscliente/mscliente/controllers/MilhasController.java
package mscliente.mscliente.controllers;

import mscliente.mscliente.DTO.MilhasDTO;
import mscliente.mscliente.DTO.TransacaoDTO;
import mscliente.mscliente.DTO.ClienteDTO;
import mscliente.mscliente.DTO.ExtratoDTO;
import mscliente.mscliente.services.MilhasService;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/clientes/{id}")
public class MilhasController {

    @Autowired
    private MilhasService service;

    @PutMapping("/milhas")
    public ResponseEntity<ClienteDTO> comprarMilhas(@PathVariable Long id, @RequestBody MilhasDTO milhas) {
        System.out.println("Comprar milhas iniciado");
        ClienteDTO response = service.comprarMilhas(id, milhas);
        return ResponseEntity.ok(response);
    }

    @GetMapping("/milhas")
    public ResponseEntity<ExtratoDTO> listarTransacoes(@PathVariable Long id) {
        ExtratoDTO extrato = service.listarTransacoesCliente(id);
        return ResponseEntity.ok(extrato);
    }

    // @GetMapping
    // public ResponseEntity<ExtratoDTO> emitirExtrato(@PathVariable Long id) {
    // ExtratoDTO extrato = service.emitirExtrato(id);
    // return ResponseEntity.ok(extrato);
    // }
}

File: backend/cliente-service/src/main/kotlin/br/ufpr/dac/cliente_service/resource/MilhasController.kt
package br.ufpr.dac.cliente_service.resource

import utils.dto.ExtratoDTO
import br.ufpr.dac.cliente_service.resource.dto.MilhasCompraDTO
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.CrossOrigin
import utils.dto.ClienteOutputDTO
import org.springframework.beans.factory.annotation.Value

@RestController
@RequestMapping("/clientes/{codigo}/milhas")
@CrossOrigin(origins = ["\${gateway.url}"])
class MilhasController(private val service: MilhasService) {

    @PutMapping
    fun comprarMilhas(@PathVariable codigo: Long, @RequestBody milhas: MilhasCompraDTO): ResponseEntity<ClienteOutputDTO> {
        val response = service.comprarMilhas(codigo, milhas)
        return ResponseEntity.ok(response)
    }

    @GetMapping
    fun emitirExtrato(@PathVariable codigo: Long): ResponseEntity<ExtratoDTO> {
        val extrato = service.emitirExtrato(codigo)
        return ResponseEntity.ok(extrato)
    }
}